### PR TITLE
chore: remove Div Int and Mod Int instances, so mathlib can use different ones

### PR DESCRIPTION
**Pull request by kim-em**
Investigating one solution to Mario's RFC about the `Int Div` convention. [zulip](https://leanprover.zulipchat.com/#narrow/stream/341532-lean4-dev/topic/.5BRFC.5D.20Int.2Ediv.20convention).

Just seeing what happens in CI for now.

### CODE REVIEW

File: src/Init/Data/Int/Basic.lean [hunk 1]
```diff
@@ -134,26 +134,51 @@ def natAbs (m : @& Int) : Nat :=
   | ofNat m   => m
   | negSucc m => m.succ
 
-@[extern "lean_int_div"]
-def div : (@& Int) → (@& Int) → Int
+/--
```
Comment (gebner) [inline src/Init/Data/Int/Basic.lean:137]:
```suggestion
/-!
```

File: src/Init/Data/Int/Basic.lean [hunk 2]
```diff
@@ -134,26 +134,51 @@ def natAbs (m : @& Int) : Nat :=
...
+* `Int.tdiv`, `Int.tmod`: T-rounding (truncating division):
+  satisfies `tdiv x y = round_to_zero (x / y)`
+
+In each case, the pair of functions unconditionally satisfies
+`x % y + (x / y) * y = x`
+which is used to fix the value of one function based on the other.
+All versions also satisfy `x / 0 = 0` and `x % 0 = x`.
+
+Lean 3 used `Int.ediv` and `Int.emod` for the typeclass instances.
+Lean 4 is now using `Int.tdiv` and `Int.tmod`,
```
Comment (gebner) [inline src/Init/Data/Int/Basic.lean:154]: This doesn't seem applicable since the PR removes these instances.

File: src/Init/Data/Int/Basic.lean [hunk 3]
```diff
@@ -134,26 +134,51 @@ def natAbs (m : @& Int) : Nat :=
...
+for compatibility with C conventions.
+See https://blog.vero.site/post/modulo for the history of these conventions.
+
+Mathlib4 intends to continue using `Int.ediv` and `Int.emod`,
+and to enable that without being prescriptive,
+we do not provide instances `Div Int` or `Mod Int` here,
+preferring to directly call the functions in the few places they are needed
+in Lean 4.
```
Comment (gebner) [inline src/Init/Data/Int/Basic.lean:162]: I don't think this kind of comment should go in a docstring.

**Review by gebner — COMMENTED** [review]
Please revert all changes to the `stage0/` directory and squash the commits.

**Comment by digama0** [comment]
I think if we are going to do this in core, we should move all three definitions to core as well and give them all a native implementation.

**Comment by gebner** [comment]
I think we've decided at the dev meeting last year that we're going to change `/` to ediv (like Python).

> we should move all three definitions to core as well and give them all a native implementation.

I would already be happy if we just harmonized the division definition so that std4 doesn't redefine it.  Adding the two other ones can be a follow-up PR if that speeds things up.

**Pull request closed without merging**